### ISSUE TITLE
Rule Proposal: `vue/no-implicit-coercion`

### ISSUE THREAD

**Issue by lozinsky** [vuejs/eslint-plugin-vue#2638]
**Please describe what the rule should do:**

It would be nice to make the core rule [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion) work in `<template>` expressions.

**What category should the rule belong to?**

[] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
  <div :data-foo="!!foo" />
  <div :data-foo="~foo.indexOf('.')" />
  <div :data-foo="+foo" />
  <div :data-foo="1 * foo" />
  <div :data-foo="'' + foo" />
  <div :data-foo="`` + foo" />
  <!-- if the `disallowTemplateShorthand` option is enabled -->
  <div :data-foo="`${foo}`" />
</template>
```
